webviz_ert/controllers/plot_view_controller.py

```diff
@@ -30,60 +38,59 @@ def _new_child(parent: WebvizErtPluginABC, plot: Dict) -> Component:
 
 
-def plot_view_controller(parent: WebvizErtPluginABC, app: dash.Dash) -> None:
-    webviz_ert.controllers.multi_response_controller(parent, app)
-    webviz_ert.controllers.multi_parameter_controller(parent, app)
+def plot_view_controller(
+    parent: WebvizErtPluginABC, app: dash.Dash, suffix: str
+) -> None:
+    if suffix == "param":
+        webviz_ert.controllers.multi_parameter_controller(parent, app)
+    elif suffix == "resp":
+        webviz_ert.controllers.multi_response_controller(parent, app)
+    else:
+        raise Exception(f"Unexpected suffix `{suffix}`")
 
     @app.callback(
-        Output(parent.uuid("plot-selection-store"), "data"),
+        Output(parent.uuid(f"plot-selection-store-{suffix}"), "data"),
         [
-            Input(parent.uuid("parameter-selection-store-param"), "modified_timestamp"),
-            Input(parent.uuid("parameter-selection-store-resp"), "modified_timestamp"),
+            Input(
+                parent.uuid(f"parameter-selection-store-{suffix}"), "modified_timestamp"
+            ),
         ],
         [
-            State(parent.uuid("parameter-selection-store-param"), "data"),
-            State(parent.uuid("parameter-selection-store-resp"), "data"),
-            State(parent.uuid("plot-selection-store"), "data"),
+            State(parent.uuid(f"parameter-selection-store-{suffix}"), "data"),
+            State(parent.uuid(f"plot-selection-store-{suffix}"), "data"),
         ],
     )
     def update_plot_selection(
         _: Any,
-        __: Any,
-        parameters: Optional[List[str]],
-        responses: Optional[List[str]],
-        current_selection: Optional[List[Mapping[str, str]]],
+        selection: Optional[List[str]],
+        current_plots: Optional[List[Mapping[str, str]]],
     ) -> List[Mapping[str, str]]:
-        parameters = [] if not parameters else parameters
-        responses = [] if not responses else responses
-        current_selection = [] if not current_selection else current_selection
-        for plot in current_selection.copy():
-            if plot["name"] not in parameters and plot["name"] not in responses:
-                current_selection.remove(plot)
-        for param in parameters:
-            new_plot = {"name": param, "type": "parameter"}
-            if new_plot not in current_selection:
-                current_selection.append(new_plot)
-        for response in responses:
-            new_plot = {"name": response, "type": "response"}
-            if new_plot not in current_selection:
-                current_selection.append(new_plot)
-
-        return current_selection
+        selection = [] if not selection else selection
+        current_plots = [] if not current_plots else current_plots
+        for plot in current_plots.copy():
+            if plot["name"] not in selection:
+                current_plots.remove(plot)
+        for selected in selection:
+            new_plot = {"name": selected, "type": _suffix_to_plot_type(suffix)}
```

Comment:
Because of the split, I don't think keeping track of the plot type is required, and the current_plot can be a simple list of strings and not a list of dicts

Reply:
ah is that what you meant above as well? i'll look into it, thanks for the hint! just assumed that the info was used here or there...

Reply:
awesome you were right, thanks for catching that!
first part (removing the type) done in d0bec85d4ad9e2
gonna look at `DataType` next